Sources/Widgets/Widgets3D/LineWidget/behavior.js

```diff
@@ -122,44 +228,23 @@ export default function widgetBehavior(publicAPI, model) {
     ) {
       return macro.VOID;
     }
-    const moveHandle = model.widgetState.getMoveHandle();
     moveHandle.setVisible(false);
     if (
       model.activeState === model.widgetState.getMoveHandle() &&
       model.widgetState.getNbHandles() === 0
     ) {
-      const handle1 = model.widgetState.getHandle1();
-      model.widgetState.setNbHandles(1);
-      handle1.setOrigin(...moveHandle.getOrigin());
-      handle1.setColor(moveHandle.getColor());
-      handle1.setScale1(moveHandle.getScale1());
-      handle1.setVisible(true);
-      model.widgetState.getHandle2().setOrigin(...moveHandle.getOrigin());
-      const SVGLayerText = model.widgetState.getText();
-      SVGLayerText.setText(model.text);
-      SVGLayerText.setOrigin(
-        calculateTextPosition(model, model.widgetState.getPositionOnLine())
-      );
+      publicAPI.placeHandle(handle1, 1);
+      handle2.setOrigin(...moveHandle.getOrigin());
     } else if (
       model.activeState === model.widgetState.getMoveHandle() &&
       model.widgetState.getNbHandles() === 1
     ) {
-      model.widgetState.setNbHandles(2);
-      const handle2 = model.widgetState.getHandle2();
-      handle2.setOrigin(...moveHandle.getOrigin());
-      handle2.setColor(moveHandle.getColor());
-      handle2.setScale1(moveHandle.getScale1());
-      handle2.setVisible(true);
+      publicAPI.placeHandle(handle2, 2);
       publicAPI.setHandleDirection();
-      const SVGLayerText = model.widgetState.getText();
-      SVGLayerText.setText(model.text);
-      SVGLayerText.setOrigin(
-        calculateTextPosition(model, model.widgetState.getPositionOnLine())
-      );
       calcTextPosWithLineAngle();
       moveHandle.setVisible(true);
     } else {
-      isDragging = true;
+      model.widgetState.setIsDragging(true);
       model.openGLRenderWindow.setCursor('grabbing');
       model.interactor.requestAnimation(publicAPI);
```

Comment:
the isDragging variable is meant to describe the user moving a fully initialized widget. GrabFocus is called when the user initializes the widget. This way the grabFocus method remains as concise as it could be